### PR TITLE
refactor(network): retrieves api versions and urls based on network store

### PR DESCRIPTION
**Pull request by ygrishajev**
Network service was bothering me a bit and I went for refactoring. The root of it is attempt to depend on a single source of truth which is a network store that already holds a selected network with all its configurations.

### CODE REVIEW

File: apps/deploy-web/env/.env.sample [hunk 1]
```diff
@@ -20,4 +21,8 @@ NEXT_PUBLIC_BASE_API_TESTNET_URL=
 NEXT_PUBLIC_API_BASE_URL=
 NEXT_PUBLIC_SENTRY_DSN=
 NEXT_PUBLIC_SENTRY_SERVER_NAME=
-NEXT_PUBLIC_GA_MEASUREMENT_ID=
\ No newline at end of file
+NEXT_PUBLIC_GA_MEASUREMENT_ID=
+
+BASE_API_MAINNET_URL=
+BASE_API_TESTNET_URL=
+BASE_API_SANDBOX_URL=
```
Comment (ygrishajev) [inline apps/deploy-web/env/.env.sample:28]: in docker compose env server and browser hosts are different

File: apps/deploy-web/src/store/networkStore.ts [hunk 2]
```diff
@@ -2,70 +2,25 @@ import axios from "axios";
...
-  }
-];
+export const networks: Network[] = cloneDeep(INITIAL_NETWORKS_CONFIG);
 
-/**
- * Get the actual versions and metadata of the available networks
- */
-export const initiateNetworkData = async () => {
-  networks = await Promise.all(
+export const initiateNetworkVersions = async () => {
```
Comment (ygrishajev) [inline apps/deploy-web/src/store/networkStore.ts:14]: ideally this shouldn't happen. We could have a dedicated components/components blocking app rendering during init. I understand that it happens in SettingsProvider, however it's opaque and one should really dig in to understand this.

File: apps/deploy-web/src/types/network.ts [hunk 3]
```diff
@@ -11,4 +13,6 @@ export type Network = {
   rpcEndpoint?: string;
   version: string | null;
   enabled: boolean;
+  apiVersion: ApiVersion;
+  marketApiVersion: ApiVersion;
```
Comment (ygrishajev) [inline apps/deploy-web/src/types/network.ts:17]: moved these here as it's really a network config, and we have selected network definition logic based on configs already.

File: apps/deploy-web/src/utils/apiUtils.ts [hunk 4]
```diff
@@ -66,82 +66,62 @@ export class ApiUrlService {
...
-  static mainnetVersion() {
-    return `${browserEnvConfig.NEXT_PUBLIC_API_BASE_URL}/v1/version/mainnet`;
-  }
-  static testnetVersion() {
-    return `${browserEnvConfig.NEXT_PUBLIC_API_BASE_URL}/v1/version/testnet`;
-  }
-  static sandboxVersion() {
-    return `${browserEnvConfig.NEXT_PUBLIC_API_BASE_URL}/v1/version/sandbox`;
+
+  static get baseApiUrl() {
```
Comment (ygrishajev) [inline apps/deploy-web/src/utils/apiUtils.ts:120]: fixes a bug introduced by me earlier - api url should vary based on the network for the above endpoints

File: apps/deploy-web/src/store/networkStore.ts (hunk 2 above)
Comment (baktun14) [reply apps/deploy-web/src/store/networkStore.ts:14]: Yea we could, but the version isn't really mandatory for the app the to render. I think we only show it in the `NetworkSelect` component.

**Review by baktun14 — APPROVED** [review]

File: apps/deploy-web/src/store/networkStore.ts (hunk 2 above)
Comment (ygrishajev) [reply apps/deploy-web/src/store/networkStore.ts:14]: Well it's used in many api urls

**Pull request merged**